### PR TITLE
feat(SQS): ADR-006 stage 2 - GRPC ingest

### PR DESCRIPTION
**Pull request by p0mvn**
Closes: #XXX

## What is the purpose of the change

This PR implements ADR-006 stage:
https://www.notion.so/osmosiszone/ADR-006-Streaming-GRPC-SQS-Ingest-4e3b2ff7d23e43e2a1f3c43adc3c26bc

It entails removing Redis as a dependency and pushing updates to SQS via GRPC.

As part of this change, `poolIngester` abstraction was renamed to `poolTransformer` because it does not ingest data into Redis anymore. Rather, it only performs data transformations on pools by instrumenting TVL data. The new name reflects this.

Config was also updated, replacing Redis parameters with GRPC.

This is a PR SQS side: https://github.com/osmosis-labs/sqs/pull/153

## Testing and Verifying

The integration with SQS had been extensively tested on mainnet.

- Pools testing comparison with TFM: https://github.com/osmosis-labs/infrastructure/actions/runs/8549394756/job/23424633198

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented?
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [ ] N/A

## Summary by CodeRabbit

- **New Features**
	- Added telemetry metrics for block processing and gRPC connection errors.
	- Introduced a new time operation to measure block processing duration.
- **Refactor**
	- Updated sidecar query server storage configuration to support gRPC service settings.
	- Refactored data ingestion to utilize new interfaces and structures, enhancing the integration with gRPC services.
	- Renamed and updated various components and methods for clarity and improved functionality, focusing on data transformation and gRPC client interactions.
- **Chores**
	- Updated import paths and method signatures across multiple files for consistency and maintainability.

### CODE REVIEW

File: ingest/sqs/domain/telemetry.go [hunk 1]
```diff
@@ -27,4 +27,21 @@ var (
...
+	SQSProcessBlockDurationMetricName = "sqs_process_block_duration"
+
+	// sqs_grpc_connection_error
+	//
+	// counter that is increased if grpc connection fails
+	//
+	// Has the following labels:
+	// * err - the error returned
+	// * height - the height of the block being processed
+	SQSGRPCConnectionErrorMetricName = "sqs_grpc_connection_error"
```
Comment (p0mvn) [inline ingest/sqs/domain/telemetry.go:46]: Note: commenting to link new alerts in this Linear task:
https://linear.app/osmosis/issue/STABI-70/instrument-new-metrics-for-stage-2-sqs-grpc-refactor

**Review by mattverse — APPROVED** [review]
utACK

File: ingest/sqs/pools/transformer/pool_transformer.go [hunk 2]
```diff
@@ -186,20 +172,9 @@ func (pi *poolIngester) ProcessPoolState(ctx sdk.Context, tx repository.Tx, bloc
...
-
-	err = pi.poolsRepository.StorePools(goCtx, tx, allPoolsParsed)
-	if err != nil {
-		return err
-	}
-
-	// persist taker fees
-	err = pi.persistTakerFees(ctx, tx, denomPairToTakerFeeMap)
-	if err != nil {
-		return err
```
[the remark this answers is not in the document]
Comment (p0mvn) [reply ingest/sqs/pools/transformer/pool_transformer.go:199]: That's right

File: ingest/sqs/domain/ingester.go [hunk 3]
```diff
@@ -39,3 +41,15 @@ type BlockPools struct {
 	// CFMMPools are the CFMM pools to be ingested.
 	CFMMPools []poolmanagertypes.PoolI
 }
+
+// GracefulSQSGRPClient is an interface that defines the methods for the graceful SQS GRPC client.
+// It handles graceful connection management. So that, if a GRPC ingest method returns status.Unavailable,
+// the GRPC client will reset the connection and attempt to recreate it before retrying the ingest method.
```
[the remark this answers is not in the document]
Comment (p0mvn) [reply ingest/sqs/domain/ingester.go:47]: It retries every block.

SQS should detect being out-of-sync and fail health-check.

Additionally, we have alerts that should go off when such an issue occurs, letting us troubleshoot

**Review by PaddyMc — APPROVED** [review]
:zap:

File: ingest/sqs/domain/ingester.go [hunk 4]
```diff
@@ -39,3 +41,15 @@ type BlockPools struct {
...
+type GracefulSQSGRPClient interface {
```
[the remark this answers is not in the document]
Comment (p0mvn) [reply ingest/sqs/domain/ingester.go]: [987a11b](https://github.com/osmosis-labs/osmosis/pull/7952/commits/987a11bab5015ba38848e0b52735166aea5a0d96)

**Pull request merged**